exchanges/kline/kline.go

```diff
@@ -130,3 +131,161 @@ func validateData(trades []order.TradeHistory) error {
 	})
 	return nil
 }
+
+// String returns numeric string
+func (i Interval) String() string {
+	return i.Duration().String()
+}
+
+// Word returns text version of Interval
+func (i Interval) Word() string {
+	return durationToWord(i)
+}
+
+// Duration returns interval casted as time.Duration for compatibility
+func (i Interval) Duration() time.Duration {
+	return time.Duration(i)
+}
+
+// Short returns short string version of interval
+func (i Interval) Short() string {
+	s := i.String()
+	if strings.HasSuffix(s, "m0s") {
+		s = s[:len(s)-2]
+	}
+	if strings.HasSuffix(s, "h0m") {
+		s = s[:len(s)-2]
+	}
+	return s
+}
+
+// durationToWord returns english version of interval
+func durationToWord(in Interval) string {
+	switch in {
+	case OneMin:
+		return "onemin"
+	case ThreeMin:
+		return "threemin"
+	case FiveMin:
+		return "fivemin"
+	case TenMin:
+		return "tenmin"
+	case FifteenMin:
+		return "fifteenmin"
+	case ThirtyMin:
+		return "thirtymin"
+	case OneHour:
+		return "onehour"
+	case TwoHour:
+		return "twohour"
+	case FourHour:
+		return "fourhour"
+	case SixHour:
+		return "sixhour"
+	case EightHour:
+		return "eighthour"
+	case TwelveHour:
+		return "twelvehour"
+	case OneDay:
+		return "oneday"
+	case ThreeDay:
+		return "threeday"
+	case Fifteenday:
+		return "fifteenday"
+	case OneWeek:
+		return "oneweek"
+	case TwoWeek:
+		return "twoweek"
+	case OneMonth:
+		return "onemonth"
+	case OneYear:
+		return "oneyear"
+	default:
+		return "notfound"
+	}
+}
+
+func TotalCandlesPerInterval(start, end time.Time, interval Interval) (out uint32) {
+	switch interval {
+	case OneMin:
+		out = uint32(end.Sub(start).Minutes())
+	case ThreeMin:
+		out = uint32(end.Sub(start).Minutes() / 3)
+	case FiveMin:
+		out = uint32(end.Sub(start).Minutes() / 5)
+	case TenMin:
+		out = uint32(end.Sub(start).Minutes() / 10)
+	case FifteenMin:
+		out = uint32(end.Sub(start).Minutes() / 15)
+	case ThirtyMin:
+		out = uint32(end.Sub(start).Minutes() / 30)
+	case OneHour:
+		out = uint32(end.Sub(start).Hours())
+	case TwoHour:
+		out = uint32(end.Sub(start).Hours() / 2)
+	case FourHour:
+		out = uint32(end.Sub(start).Hours() / 4)
+	case SixHour:
+		out = uint32(end.Sub(start).Hours() / 6)
+	case EightHour:
+		out = uint32(end.Sub(start).Hours() / 8)
+	case TwelveHour:
+		out = uint32(end.Sub(start).Hours() / 12)
+	case OneDay:
+		out = uint32(end.Sub(start).Hours() / 24)
+	case ThreeDay:
+		out = uint32(end.Sub(start).Hours() / 72)
+	case Fifteenday:
+		out = uint32(end.Sub(start).Hours() / (24 * 15))
+	case OneWeek:
+		out = uint32(end.Sub(start).Hours()) / (24 * 7)
+	case TwoWeek:
+		out = uint32(end.Sub(start).Hours() / (24 * 14))
+	case OneMonth:
+		// out = uint32(end.Sub(start).Minutes / (24*))
```

Comment:
Ahh probably handle this or error?

Reply:
yeah open to feedback on the best way to handle one month because is 1 month 28 days 29 days 30 days 31 days :D

Reply:
For platforms which support 1 month, a month isn't a fixed value but starts on the 1st day of month. Are there any exchanges which support fetching 1 month candles?

Reply:
Binance and the Huobi are the only two exchanges that have 1 month candles